### PR TITLE
Create an interface for base SNS velocity IK solver

### PR DESCRIPTION
**Pull request by robodreamer**
This commit adds an interface that allows SNS Base IK
solver to be used as one of SNS IK solver types.
SNS_Base type has been defined and position and velocity
IK solver tests now include tests for this new type.

### CODE REVIEW

File: sns_ik_lib/src/sns_vel_ik_base.cpp [hunk 1]
```diff
@@ -294,7 +294,7 @@ SnsIkBase::ExitCode SnsVelIkBase::solve(const Eigen::MatrixXd& J, const Eigen::V
     return ExitCode::InternalError;
   }
   else if (*taskScaleCS < MINIMUM_FINITE_SCALE_FACTOR) {
-    ROS_WARN("Secondary goal is infeasible! scaling --> zero");
+    ROS_DEBUG_THROTTLE(0.5, "Secondary goal is infeasible! scaling --> zero");
```
Comment (chris-smith) [inline sns_ik_lib/src/sns_vel_ik_base.cpp]: If you're already making this debug, do you still need to throttle it? same question for other similar change

File: sns_ik_lib/src/sns_vel_ik_base_interface.cpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    // if the tasks include a nullspace bias task
+    taskScaleCS = 1.0;
+    dqCS = sot[1].desired;
+  }
+
+  // set box constraints
+  baseIkSolver->setBounds(dqLow, dqUpp);
+
+  // initialize the base IK solver
+  // ROS_INFO_STREAM("dqLow: " << dqLow.transpose().format(EigArrFmt));
```
Comment (chris-smith) [inline sns_ik_lib/src/sns_vel_ik_base_interface.cpp]: remove logging?

File: sns_ik_lib/src/sns_vel_ik_base_interface.cpp [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    scaleFactors[0] = taskScale;
+  }
+  else {
+    exitCode = baseIkSolver->solve(J, dx, dqCS, &dqSol, &taskScale, &taskScaleCS);
+    scaleFactors[0] = taskScale;
+    scaleFactors[1] = taskScaleCS;
+  }
+
+
+  // ROS_INFO_STREAM("taskScale: " << taskScale);
```
Comment (chris-smith) [inline sns_ik_lib/src/sns_vel_ik_base_interface.cpp]: and here

**Comment by chris-smith** [comment]
Mostly wondering about renaming the new class? `SNSVelocityBaseIK` which inherits from `SNSVelocityIK` and has a member `SnsVelIkBase`... I have a hard time figuring out what each does from the names and not getting them mixed up.

File: sns_ik_lib/src/sns_vel_ik_base_interface.cpp (hunk 3 above)
Comment (robodreamer) [reply sns_ik_lib/src/sns_vel_ik_base_interface.cpp]: I forgot to remove them..

File: sns_ik_lib/src/sns_vel_ik_base.cpp (hunk 1 above)
Comment (robodreamer) [reply sns_ik_lib/src/sns_vel_ik_base.cpp]: Right. I agree. Debug will be fine.

Comment (robodreamer) [reply sns_ik_lib/src/sns_vel_ik_base.cpp]: Thanks for your comments! :)

**Review by chris-smith — APPROVED** [review]
lgtm

**Pull request merged**